pgmpy/factors/Factor.py

```diff
@@ -194,7 +194,7 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-        marginalize_index = np.array(np.where(np.in1d(factor.scope(), variables)))
+        marginalize_index = np.array(np.where(np.array([i in variables for i in factor.scope()])))
```

Comment:
`marginalize_index` is only being used for indexing, right? So in the interest of better performance, simply doing
```python
marginalize_index = np.array([i in variables for i in factor.scope()])
```
will create a boolean array which can itself be used for indexing. No need for the extra `np.where` and `np.array` calls.